src/main/java/com/gojek/beast/factory/BeastFactory.java

```diff
@@ -89,23 +97,48 @@ public BeastFactory(AppConfig appConfig, BackOffConfig backOffConfig, ProtoMappi
 
     public Sink createBigQuerySink() {
         BigQuery bq = getBigQueryInstance();
-        Sink bqSink = new BqSink(bq, TableId.of(appConfig.getDataset(), appConfig.getTable()));
+        BQResponseParser responseParser = new BQResponseParser();
+        BQErrorHandler bqErrorHandler = createOOBErrorHandler();
+        Sink bqSink = new BqSink(bq, TableId.of(appConfig.getDataset(), appConfig.getTable()), responseParser, bqErrorHandler);
         return new RetrySink(bqSink, new ExponentialBackOffProvider(backOffConfig.getExponentialBackoffInitialTimeInMs(), backOffConfig.getExponentialBackoffMaximumTimeInMs(), backOffConfig.getExponentialBackoffRate(), new BackOff()), appConfig.getMaxPushAttempts());
     }
 
+    public BQErrorHandler createOOBErrorHandler() {
+        final Storage gcsStore = getGCStorageInstance();
+        ErrorWriter errorWriter;
+        if (appConfig.isGCSErrorSinkEnabled()) {
+            final String bucketName = appConfig.getGcsBucket();
+            final String basePathPrefix = bucketName + "/" + appConfig.getGcsPathPrefix();
+            errorWriter = new GCSErrorWriter(gcsStore, bucketName, basePathPrefix);
+        } else {
+            errorWriter = new DefaultLogWriter();
```

Comment:
1. Could we rename

```
OOBErrorHandler
DT_PREFIX  // you can have longer name
```

2. You could assign defaultLogWriter and if sinkenabled override it

Reply:
1. Renamed DT_PREFIX. OOBErrorHandler was the choice from the team as it pertains to handling just OOB.
2. Done.